File: Demo/MNG-Ads-SDK/SettingsViewController.m
//
//  SettingsViewController.m
//  MNG-Ads-SDK-Demo
//
//  Created by Bensalah Med Amine on 16/12/2016.
//  Copyright © 2016 MAdvertise. All rights reserved.
//

#import "SettingsViewController.h"
@import MAdvertiseCMP;
#import <AdSupport/ASIdentifierManager.h>
@interface SettingsViewController  ()
@property (weak, nonatomic) IBOutlet UITextField *consentFlagTextfiled;

@end

@implementation SettingsViewController {
    AVPlayer *avPlayerq;
    AVPlayerLayer *videoLayer ;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.appidTextField.text = MNG_ADS_APP_ID;
    self.debugSwitch.on = DEBUG_MODE_ENABLED;
    _tcfTextView.delegate = self;
    _idfaTextView.delegate = self ;
    
    NSNumber * consentFlag = MNG_ADS_ConsentFlag ;
    _consentFlagTextfiled.text = [NSString stringWithFormat:@"%@",[consentFlag stringValue] ] ;
    _tcfTextView.text = [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_TCString"];
    _idfaTextView.text = [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
    _GoogleConsentTextview.text = [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_AddtlConsent"];
    [self getExternalPurposes];
    [self getPurposes];
    [self debugAllIabTcfKeyCache] ;
    _cmpLanguageTextField.text = CMP_LANGUAGE ;
    CMPConsentManager.sharedInstance.delegate = self;
   
    
    _vendorsTexTfield.text = @"746,755";
    _externalVendorsTextfield.text = @"8,9";
    _idExtPurposetextField.text = @"5";
    
    
}
-(void)viewWillAppear:(BOOL)animated{
    NSString  * madvertiseconsentProvided = [[NSUserDefaults standardUserDefaults]objectForKey:@"madvertiseconsentProvided"];
    [Utils displayToastWithMessage:madvertiseconsentProvided];
}
-(void)viewDidAppear:(BOOL)animated {
    APP_DELEGATE.drawerViewController.statusBarStyle = UIStatusBarStyleLightContent;
}

-(void)viewDidDisappear:(BOOL)animated{
    [super viewDidDisappear:animated];
    [[NSNotificationCenter defaultCenter] removeObserver:self];

}

- (void)consentManagerRequestsToShowConsentTool:(CMPConsentManager * _Nonnull)consentManager forVendorList:(CMPVendorList * _Nonnull)vendorList {
    if ([CMPConsentManager.sharedInstance showConsentToolFromController:self withPopup:YES]) {
    }
}

-(void)tcfOnConsentStringDidChange:(TCFString *)newTcfConsentString consentProvided:(NSString *)consentProvided {
    NSLog(@"------> Consent String changed : %@", newTcfConsentString.tcfString);
//    NSLog(@"------> consentProvided changed : %@", consentProvided);

    [Utils displayToastWithMessage:consentProvided];

    _tcfTextView.text = [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_TCString"];
    _GoogleConsentTextview.text = [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_AddtlConsent"];
    [self getExternalPurposes];
    [self debugAllIabTcfKeyCache] ;
    [self getPurposes];
    
}

-(void)didAcceptAllTcfConsentString{
     [Utils displayToastWithMessage:@"didAcceptAllTcfConsentString"];
    
}
-(void)didRefuseAllTcfConsentString{
     [Utils displayToastWithMessage:@"didRefuseAllTcfConsentString"];
    
}
-(void)consentManagerDidFailWithErrorWithError:(NSError *)error{
    [Utils displayToastWithMessage:error.localizedDescription];
    NSLog(@"------> consentManagerDidFailWithErrorWithError : %@", error.localizedDescription);

}
- (void)consentManagerRequestsToPresentPrivacyPolicyWithUrl:(NSString*)url{
    [Utils displayToastWithMessage:url];
}
- (IBAction)gdprsettings:(UIButton*)sender {
    if (sender.tag == 0) {
          if ([CMPConsentManager.sharedInstance showConsentToolFromController:self withPopup:NO]) {
            APP_DELEGATE.drawerViewController.statusBarStyle = UIStatusBarStyleLightContent;
        }
      
    } else if (sender.tag == 1) {
          if ([CMPConsentManager.sharedInstance showConsentToolFromController:self withPopup:YES]) {
                    APP_DELEGATE.drawerViewController.statusBarStyle = UIStatusBarStyleLightContent;

        }
    }
    
}


- (IBAction)OpenMenu:(id)sender {
    [[APP_DELEGATE drawerViewController]openMenu];
}
- (IBAction)submit:(id)sender {
    [self.appidTextField endEditing:YES];
    [self.cmpLanguageTextField endEditing:YES];
    [[NSUserDefaults standardUserDefaults]setObject:[NSNumber numberWithInt:[_consentFlagTextfiled.text intValue]]
                                             forKey:@"ConsetFlagBlueSTack"];
    [[NSUserDefaults standardUserDefaults]synchronize];
    NSString *newAppId = self.appidTextField.text;
    if (!newAppId || [newAppId isEqualToString:@""]) {
        NSLog(@"put a valid appId");
        return;
    }
    
  
    NSString * oldAppId = [[NSUserDefaults standardUserDefaults] objectForKey: @"SavedappId" ];
    if (![newAppId isEqualToString:oldAppId]) {
        [[NSUserDefaults standardUserDefaults]setObject:newAppId
                                                 forKey:@"SavedappId"];
        [[NSUserDefaults standardUserDefaults]synchronize];
        [MNGAdsSDKFactory setDelegate:self];
        [MNGAdsSDKFactory initWithAppId:newAppId];
    }
    [[NSUserDefaults standardUserDefaults]setObject:@(self.debugSwitch.on)
                                             forKey:@"SavedDebug"];
    [[NSUserDefaults standardUserDefaults]setObject:_cmpLanguageTextField.text
                                             forKey:@"CMP_LANGUAGE"];
    [[NSUserDefaults standardUserDefaults]synchronize];
    [MNGAdsSDKFactory setDebugModeEnabled:DEBUG_MODE_ENABLED];
    
}

-(void)MNGAdsSDKFactoryDidFinishInitializing{
    NSLog(@"MNGAds sucess initialization");
  
    [Utils displayToastWithMessage:@"MNGAds sucess initialization"];

}

-(void)MNGAdsSDKFactoryDidFailInitializationWithError:(NSError *)error {
    NSLog(@"MNGAds failed initialization");
    [Utils displayToastWithMessage:@"MNGAds failed initialization"];
}

- (IBAction)AcceptAllPressed:(id)sender {
    [CMPConsentManager.sharedInstance acceptAllConsentTool];
        NSLog(@"------> Consent String ALL Accepted : %@",  [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_TCString"]);
    
     _GoogleConsentTextview.text = [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_AddtlConsent"];
     [self getExternalPurposes];
   [self debugAllIabTcfKeyCache] ;
    [self getPurposes];

    
}
- (IBAction)refuseAllPressed:(id)sender {
    [CMPConsentManager.sharedInstance refuseAllConsentTool];
     NSLog(@"------> IABTCF_TCString : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_TCString"]);
     _GoogleConsentTextview.text = [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_AddtlConsent"];
     [self getExternalPurposes];
    [self debugAllIabTcfKeyCache] ;
    [self getPurposes];
}


-(void)debugAllIabTcfKeyCache{
    
    NSLog(@"------> IABTCF_CmpSdkID : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_CmpSdkID"]);
    
    
    NSLog(@"------> IABTCF_CmpSdkVersion : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_CmpSdkVersion"]);
    
    NSLog(@"------> IABTCF_PolicyVersion : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PolicyVersion"]);
    
     NSLog(@"------> IABTCF_gdprApplies : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_gdprApplies"]);
    
     NSLog(@"------> IABTCF_PublisherCC : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PublisherCC"]);
    
     NSLog(@"------> IABTCF_PurposeOneTreatment : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PurposeOneTreatment"]);
    
     NSLog(@"------> IABTCF_UseNonStandardStacks : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_UseNonStandardStacks"]);
    
     NSLog(@"------> IABTCF_VendorConsents : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_VendorConsents"]);
    
     NSLog(@"------> IABTCF_VendorLegitimateInterests : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_VendorLegitimateInterests"]);
    
    NSLog(@"------> IABTCF_PurposeConsents : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PurposeConsents"]);
    
    NSLog(@"------> IABTCF_PurposeLegitimateInterests : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PurposeLegitimateInterests"]);
    
    NSLog(@"------> IABTCF_SpecialFeaturesOptIns : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_SpecialFeaturesOptIns"]);
    
    NSLog(@"------> IABTCF_PublisherConsent : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PublisherConsent"]);
    
     NSLog(@"------> IABTCF_PublisherRestrictions1 : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PublisherRestrictions1"]);
    
     NSLog(@"------> IABTCF_PublisherRestrictions2 : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PublisherRestrictions2"]);
    
     NSLog(@"------> IABTCF_PublisherRestrictions3 : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PublisherRestrictions3"]);
    
     NSLog(@"------> IABTCF_PublisherRestrictions4 : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PublisherRestrictions4"]);
    
    NSLog(@"------> IABTCF_PublisherRestrictions5 : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_PublisherRestrictions5"]);
    
    NSLog(@"------> IABTCF_Madvertise_ExternalVendors : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_Madvertise_ExternalVendors"]);
    
     NSLog(@"------> IABTCF_TCString : %@", [[NSUserDefaults standardUserDefaults]objectForKey:@"IABTCF_TCString"]);

}


- (void)getExternalPurposes {
    NSArray * purposeIDs =    [CMPConsentManager.sharedInstance getExternalPurposesIDs];
    
    _externalPurposeId.text = [NSString stringWithFormat:@"%@",[purposeIDs description]];
}

- (IBAction)showExternalPurposes:(id)sender {
    [self getExternalPurposes];
}
- (IBAction)showPurposeIDs:(id)sender {
    [self getPurposes];

}

- (void)getPurposes {
    NSArray * purposeIDs =    [CMPConsentManager.sharedInstance getPurposesIDs];

    _purposeTextField.text = [NSString stringWithFormat:@"%@",[purposeIDs description]];
    
}
- (NSArray*)getIdsFromString:(NSString *)input {
    NSMutableArray * idArray = [[NSMutableArray alloc]init];
    if (input.length != 0 ){
        NSArray *arrayOfComponents = [input componentsSeparatedByString:@","];
        if (arrayOfComponents.count != 0) {
            for (int i = 0 ; i < arrayOfComponents.count; i++) {
                if ([arrayOfComponents[i] intValue] != 0) {
                    [idArray addObject:[NSNumber numberWithInt:[arrayOfComponents[i] intValue]]];
                }
            }
        }
    }
    
    
    return idArray;
}

- (IBAction)updateExtenalPurposes:(id)sender {
    
    NSArray *arrayOfvendor=[self getIdsFromString:_vendorsTexTfield.text];
    NSArray *arrayOfExternalVendor=[self getIdsFromString:_externalVendorsTextfield.text];
    NSArray *arrayOfExternalPurpose =[self getIdsFromString:_idExtPurposetextField.text];

    [CMPConsentManager.sharedInstance updateExternalPurposesIDs:arrayOfExternalPurpose vendors:arrayOfvendor externalvendors:arrayOfExternalVendor type:_updateexternalSwitch.isOn];
    [self debugAllIabTcfKeyCache];

    [self.view endEditing:YES];

}


@end

File: DemoSwift/DemoSwift/Settings - CMP/SettingsViewController.swift
//
//  SettingsViewController.swift
//  BlueStackDemoDev
//
//  Created by MedSghaier on 19/8/2022.
//

import UIKit
import MAdvertiseCMP
import BlueStackSDK

class SettingsViewController: UIViewController, CMPConsentManagerDelegate , MNGAdsSDKFactoryDelegate {
    
    
    @IBOutlet weak var appIdTextField: UITextField!
    
    
    @IBOutlet weak var consentFlagTextfield: UITextField!
    
    
    
    @IBOutlet weak var cmpLanguageTextField: UITextField!
    
    
    @IBOutlet weak var debugSwitch: UISwitch!
    
    
    @IBOutlet weak var updateExternalSwitch: UISwitch!
    
    
    @IBOutlet weak var idExtPurposeTextField: UITextField!
    
    
    @IBOutlet weak var externalVendorsTextField: UITextField!

    
    @IBOutlet weak var vendorsTextField: UITextField!
    
    
    static let consentStringKey = "IABTCF_TCString"
    static let madvertiseConsentStringKey = "MadvertiseConsent_ConsentString"
    
    var storedConsentString: String?
    var storedMadvertiseConsentString: String?
    

    @IBOutlet weak var tcfTextView: UITextView!
    
    @IBOutlet weak var idfaTextView: UITextView!
    @IBOutlet weak var googleTextView: UITextView!
    
    @IBOutlet weak var externalpurposesTextView: UITextView!
    
    
    @IBOutlet weak var purposeTextView: UITextView!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setUpTextFields()
        
        CMPConsentManager.shared.delegate = self
        
        // Configure the CMPConsentManager shared instance.
        let cmpLanguage = CMPLanguage(string: "fr") ?? CMPLanguage.DEFAULT_LANGUAGE
        
        CMPConsentManager.shared.configure("MAdvertiseCMPSettingsTCFV2_config_fr", language: cmpLanguage, appId: DemoSwiftConstants.appID, publisherCC: "FR",autoClose:true)
        
        
        updateTextViews()
        self.idExtPurposeTextField.text = "5";
        self.externalVendorsTextField.text = "8,9";
        self.vendorsTextField.text = "746,755";
       
        
        //gesture recognizer
        let tap = UITapGestureRecognizer(target: self, action: #selector(self.handleTap(_:)))
        self.view.addGestureRecognizer(tap)
    }
    
    @objc func handleTap(_ sender: UITapGestureRecognizer? = nil) {
        // handling code
        self.view.endEditing(true)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        let madvertiseconsentProvided = UserDefaults.standard.object(forKey: "madvertiseconsentProvided") as? String
        APP_DELEGATE.displayToast(message: madvertiseconsentProvided ?? "")
    }
    
    //MARK: -setup textfields
    func setUpTextFields()  {
        //app Id
        self.appIdTextField.text = MNG_ADS_APP_ID
        //Debug Mode
        let debugOn = UserDefaults.standard.bool(forKey: "SavedDebug")
        self.debugSwitch.isOn = debugOn
        
        //Consent number
        self.consentFlagTextfield.text = String(format: "%d", MNG_ADS_ConsentFlag ?? 1)
        //CMP Language
        self.cmpLanguageTextField.text = MNG_ADS_CMPLanguage
        
        
    }
    
    fileprivate func updateTextViews() {
        tcfTextView.text = UserDefaults.standard.object(forKey: "IABTCF_TCString") as? String
        googleTextView.text = UserDefaults.standard.object(forKey: "IABTCF_AddtlConsent") as? String
        idfaTextView.text = UUID().uuidString
        getPurposes()
        getExternalPurpose()
        debugAllIabTcfKeyCache()
    }
    
    func getExternalPurpose()  {
        let purposeArray = CMPConsentManager.shared.getExternalPurposesIDs() as [Int]
        externalpurposesTextView.text = purposeArray.description
        
    }
    
    func getPurposes(){
        let purposeIds = CMPConsentManager.shared.getPurposesIDs() as [Int]
        purposeTextView.text = purposeIds.description
    }
    
    
    // MARK: - CMPConsentManagerDelegate
    
    func consentManagerRequestsToShowConsentTool(_ consentManager: CMPConsentManager, forVendorList vendorList: CMPVendorList) {
        NSLog("CMP Requested ConsentTool Display");
        
        // You should display the consent tool UI, when user is ready…
        let _ = consentManager.showConsentTool(fromController: self, withPopup: true)
        
        // Since the vendor list is provided in parameter of this delegate method, you can also build your own UI to ask for
        // user consent and simply save the resulting consent string in the relevant IAB keys (see the IAB specification for
        // more details about this).
        //
        // To generate a valid IAB consent string easily, you can use the CMPConsentString class.
    }
    
    func tcfOnConsentStringDidChange(_ newTcfConsentString: TCFString, consentProvided: String) {
        print("-------->TCFString did change")
        APP_DELEGATE.displayToast(message: consentProvided)
        updateTextViews()
    }
    
    
    func consentManagerDidFailWithError(error: Error) {
        APP_DELEGATE.displayToast(message: error.localizedDescription)
        NSLog("consentManagerDidFailWithErrorWithError \(error.localizedDescription)")
    }
    
    func consentManagerRequestsToPresentPrivacyPolicy(url: String) {
        APP_DELEGATE.displayToast(message: url)
    }
    // MARK: Show consent tool action
    
    @IBAction func showConsentToolButtonTapped(_ sender: UIButton) {
        // Showing the consent tool manually
        switch sender.tag {
        case 0:
            if !CMPConsentManager.shared.showConsentTool(fromController: self, withPopup: false) {
                
                // Note: showing the consent tool might fail for several reasons (check the API documentation for more information).
                // For better user experience, it is advised to display an error if the consent tool can't be opened when it has been
                // requested by the user.
                let alert = UIAlertController(title: "Service unavailable",
                                              message: "Setting your privacy preferences is not possible at the moment, please try again later",
                                              preferredStyle: .alert)
                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
                self.present(alert, animated: true)
                
            }
        case 1:
            if !CMPConsentManager.shared.showConsentTool (fromController: self, withPopup: true){
                
                // Note: showing the consent tool might fail for several reasons (check the API documentation for more information).
                // For better user experience, it is advised to display an error if the consent tool can't be opened when it has been
                // requested by the user.
                let alert = UIAlertController(title: "Service unavailable",
                                              message: "Setting your privacy preferences is not possible at the moment, please try again later",
                                              preferredStyle: .alert)
                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
                self.present(alert, animated: true)
                
            }
            
        default:
            break
        }
        
    }
    
    @IBAction func acceptAllBtnPressed(_ sender: Any) {
        CMPConsentManager.shared.acceptAllConsentTool()
        updateTextViews()
        APP_DELEGATE.displayToast(message: "didAcceptAllTcfConsentString")
    }
    @IBAction func refuseAllBtnPressed(_ sender: Any) {
        CMPConsentManager.shared.refuseAllConsentTool()
        updateTextViews()
        APP_DELEGATE.displayToast(message: "didRefuseAllTcfConsentString")
    }
    @IBAction func showExternalPurposeBtnPressed(_ sender: Any) {
        getExternalPurpose()
    }
    
    
    
    @IBAction func showPurposesIDs(_ sender: UIButton) {
        self.getPurposes()
    }
    
    
    
    @IBAction func updateExternalPurposes(_ sender: UIButton) {
        
        let arrayOfExternalPurpose = self.getIdsFromString(input: self.idExtPurposeTextField.text!)
        let arrayOfExternalVendors = self.getIdsFromString(input: self.externalVendorsTextField.text!)
        let arrayOfVendors = self.getIdsFromString(input: self.vendorsTextField.text!)
        
        
        CMPConsentManager.shared.updateExternalPurposesIDs(arrayOfExternalPurpose, vendors: arrayOfVendors, externalvendors: arrayOfExternalVendors, type: self.updateExternalSwitch.isOn)
        self.debugAllIabTcfKeyCache()
        self.view.endEditing(true)
        
    }
    
    //MARK: - get Ids from textfields
    func getIdsFromString(input: String) ->[Int] {
        var result: [Int] = []
        if input.count > 0 {
            let arrayOfComponents = input.components(separatedBy: ",")
            for id in arrayOfComponents {
                result.append(Int(id)!)
            }
        }
        return result
    }
    
    func debugAllIabTcfKeyCache(){
        
        
        print("------> IABTCF_CmpSdkID :\(UserDefaults.standard.object(forKey: "IABTCF_CmpSdkID") ?? "IABTCF_CmpSdkID not found")");
        
        
        print("------> IABTCF_CmpSdkVersion :\(UserDefaults.standard.object(forKey: "IABTCF_CmpSdkVersion") ?? "IABTCF_CmpSdkVersion not found")");
        
        print("------> IABTCF_PolicyVersion :\(UserDefaults.standard.object(forKey: "IABTCF_PolicyVersion") ?? "IABTCF_PolicyVersion not found")");
        
        print("------> IABTCF_gdprApplies :\(UserDefaults.standard.object(forKey: "IABTCF_gdprApplies") ?? "IABTCF_gdprApplies not found")");
        
        print("------> IABTCF_PublisherCC :\(UserDefaults.standard.object(forKey: "IABTCF_PublisherCC") ?? "IABTCF_PublisherCC not found")");
        
        print("------> IABTCF_PurposeOneTreatment :\(UserDefaults.standard.object(forKey: "IABTCF_PurposeOneTreatment") ?? "IABTCF_PurposeOneTreatment not found")");
        
        print("------> IABTCF_UseNonStandardStacks :\(UserDefaults.standard.object(forKey: "IABTCF_UseNonStandardStacks") ?? "IABTCF_UseNonStandardStacks not found")");
        
        
        print("------> IABTCF_VendorConsents :\(UserDefaults.standard.object(forKey: "IABTCF_VendorConsents") ?? "IABTCF_VendorConsents not found")");
        
        
        print("------> IABTCF_VendorLegitimateInterests :\(UserDefaults.standard.object(forKey: "IABTCF_VendorLegitimateInterests") ?? "IABTCF_VendorLegitimateInterests not found")");
        
        
        print("------> IABTCF_PurposeConsents :\(UserDefaults.standard.object(forKey: "IABTCF_PurposeConsents") ?? "IABTCF_PurposeConsents not found")");
        
        
        print("------> IABTCF_PurposeLegitimateInterests :\(UserDefaults.standard.object(forKey: "IABTCF_PurposeLegitimateInterests") ?? "IABTCF_PurposeLegitimateInterests not found")");
        
        print("------> IABTCF_SpecialFeaturesOptIns :\(UserDefaults.standard.object(forKey: "IABTCF_SpecialFeaturesOptIns") ?? "IABTCF_SpecialFeaturesOptIns not found")");
        
        print("------> IABTCF_PublisherConsent :\(UserDefaults.standard.object(forKey: "IABTCF_PublisherConsent") ?? "IABTCF_PublisherConsent not found")");
        
        print("------> IABTCF_PublisherRestrictions1 :\(UserDefaults.standard.object(forKey: "IABTCF_PublisherRestrictions1") ?? "IABTCF_PublisherRestrictions1 not found")");
        
        
        print("------> IABTCF_PublisherRestrictions2 :\(UserDefaults.standard.object(forKey: "IABTCF_PublisherRestrictions2") ?? "IABTCF_PublisherRestrictions2 not found")");
        
        print("------> IABTCF_PublisherRestrictions3 :\(UserDefaults.standard.object(forKey: "IABTCF_PublisherRestrictions3") ?? "IABTCF_PublisherRestrictions3 not found")");
        
        
        print("------> IABTCF_PublisherRestrictions4 :\(UserDefaults.standard.object(forKey: "IABTCF_PublisherRestrictions4") ?? "IABTCF_PublisherRestrictions4 not found")");
        
        print("------> IABTCF_PublisherRestrictions5 :\(UserDefaults.standard.object(forKey: "IABTCF_PublisherRestrictions5") ?? "IABTCF_PublisherRestrictions5 not found")");
        
        
        print("------> IABTCF_Madvertise_ExternalVendors :\(UserDefaults.standard.object(forKey: "IABTCF_Madvertise_ExternalVendors") ?? "IABTCF_Madvertise_ExternalVendors not found")");
        
        print("------> IABTCF_TCString :\(UserDefaults.standard.object(forKey: "IABTCF_TCString") ?? "IABTCF_TCString not found")");
        
    }
    
    // MARK: - Menu open
    
    @IBAction func openMenu(_ sender: Any) {
        APP_DELEGATE.drawerViewController?.openMenu()
    }
    
    
    
    @IBAction func submit(_ sender: Any) {
        self.appIdTextField.endEditing(true)
        self.cmpLanguageTextField.endEditing(true)
        
        //set the new value of consent
        let consentFlagValue = Int(self.consentFlagTextfield.text ?? "")
        UserDefaults.standard.set(consentFlagValue, forKey: ConsetFlagBlueSTack)
        //debug mode
        UserDefaults.standard.set(self.debugSwitch.isOn, forKey: "SavedDebug")
        MNGAdsSDKFactory.getVersion()
        MNGAdsSDKFactory.setDebugModeEnabled(self.debugSwitch.isOn)
        //cmp language
        UserDefaults.standard.set(self.cmpLanguageTextField.text ?? "fr", forKey: CMP_LANGUAGE)
        //APPID
        let newAppId = self.appIdTextField.text
        guard let newAppId = newAppId else {
            return
        }
        if newAppId == "" {
            NSLog("put a valid appId");
            return;
        }
        
        let oldAppId  = UserDefaults.standard.object(forKey: SavedAppId) as? String
        
        if newAppId != oldAppId {
            MNG_ADS_APP_ID = newAppId
            MNGAdsSDKFactory.setDelegate(self)
            MNGAdsSDKFactory.initWithAppId(newAppId)
        }
        
        UserDefaults.standard.synchronize()
    }
    
    //MARK: -MNGADS SDK Factory delegate
    func mngAdsSDKFactoryDidFinishInitializing() {
        NSLog("MNGAds sucess initialization");
        APP_DELEGATE.displayToast(message: "MNGAds sucess initialization")
    }
    
    func mngAdsSDKFactoryDidFailInitializationWithError(_ error: Error!) {
        NSLog("MNGAds failed initialization");
        APP_DELEGATE.displayToast(message: "MNGAds failed initialization");
    }
    
    
    
}
